### PR TITLE
Implement synchronization with mlflow

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: cli/data_loader.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                existing_run_uuids = set([
+                    str(e.get_properties().get(DataLoader.MLFLOW_RUN_ID_PROPERTY)) for e in existing_experiments
+                ])
+
+                for run_info in run_infos:
+                    run_qualified_name = self._get_run_qualified_name(experiment, run_info)
+                    if run_info.run_uuid not in existing_run_uuids:
+                        click.echo("Loading run {}".format(run_qualified_name))
+                        run = mlflow_client.get_run(run_info.run_uuid)
+                        exp_uuid = self._create_neptune_experiment(experiment, run)
```
Comment (dominik-jastrzebski) [inline cli/data_loader.py]: The state of the experiment will be `running` forever?

File: tests/cli/test_data_loader.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        # given
+        exp = MagicMock()
+        exp.name = 'exp_name'
+        run_info = MagicMock()
+        run_info.name = None
+        run_info.run_uuid = 'run_uuid'
+
+        # then
+        # pylint: disable=protected-access
+        self.assertEqual(DataLoader._get_run_qualified_name(exp, run_info), "exp_name/run_uuid")
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: Either rename `then` to `expect` or extract `DataLoader._get_run_qualified_name(exp, run_info)` to the `when` section.

File: tests/cli/test_data_loader.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    def test_get_metric_file(self):
+        # given
+        exp = MagicMock()
+        exp.experiment_id = 'exp_id'
+        run_info = MagicMock()
+        run_info.run_uuid = 'run_uuid'
+        metric_key = 'metric_key'
+
+        # then
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: same as above

File: tests/cli/test_data_loader.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    def test_get_tags(self):
+        # given
+        exp = MagicMock()
+        exp.name = 'EXPeriMENT-NaMe'
+        run = MagicMock()
+        run.info = MagicMock()
+        run.info.name = 'RuN-NAme'
+
+        # then
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: same as above

File: tests/cli/test_data_loader.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        exp.experiment_id = 123
+        exp.name = 'EXPeriMENT-NaMe'
+        run = MagicMock()
+        run.info = MagicMock()
+        run.info.run_uuid = str(uuid.uuid4())
+        run.info.name = 'Cool run name'
+        run.data = MagicMock()
+        run.data.tags = [Tag(key='key1', value='value1'), Tag(key='key2', value='value2')]
+
+        # then
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: same as above

File: tests/cli/test_data_loader.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        )
+
+    def test_get_params(self):
+        # given
+        Param = namedtuple('Param', 'key, value')
+        run = MagicMock()
+        run.data = MagicMock()
+        run.data.params = [Param(key='key1', value='value1'), Param(key='key2', value='value2')]
+
+        # then
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: same as above

File: tests/cli/test_data_loader.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        # pylint: disable=protected-access
+        self.assertEqual(
+            DataLoader._get_tags(exp, run),
+            ['experiment-name', 'run-name'])
+
+    def test_get_properties(self):
+        # given
+        Tag = namedtuple('Tag', 'key, value')
+        exp = MagicMock()
+        exp.experiment_id = 123
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: You can split the code into multiple sections to make it more readable. This applies to other tests too, but they are shorter and it doesn't seem obvious there.

```
Tag = namedtuple('Tag', 'key, value')

# given
exp = MagicMock()
exp.experiment_id = 123
exp.name = 'EXPeriMENT-NaMe'

# and
run = MagicMock()

# and
run.info = MagicMock()
run.info.run_uuid = str(uuid.uuid4())
run.info.name = 'Cool run name'

# and
run.data = MagicMock()
run.data.tags = [Tag(key='key1', value='value1'), Tag(key='key2', value='value2')]
```

File: cli/data_loader.py (hunk 1 above)
Comment (aniezurawski) [reply cli/data_loader.py]: Experiment is created via `with` so context manager will set it to `succeeded`.

File: tests/cli/test_data_loader.py [hunk 8]
```diff
@@ -28,23 +28,29 @@ def test_get_run_qualified_name(self):
...
+        # expect
         # pylint: disable=protected-access
         self.assertEqual(DataLoader._get_run_qualified_name(exp, run_info), "exp_name/run_uuid")
 
     def test_get_metric_file(self):
         # given
         exp = MagicMock()
         exp.experiment_id = 'exp_id'
+
+        #and
```
Comment (dominik-jastrzebski) [inline tests/cli/test_data_loader.py]: Replace all `#and` with `# and`.

**Pull request merged**